### ISSUE TITLE
Programmatic configuration of 'maxArchiveFiles' parameter

### ISSUE THREAD

**Issue by DirkDeclercq** [NLog/NLog#3715]
NLog 4.6.8

Just like I did with the 'fileName' parameter I'm trying to configure the 'maxArchiveFiles' (see config-below)
In code I have the following : (at startup)
'''
                using (var disableScope = LogManager.DisableLogging())
                {
                    var logFilePath = optionsService.GetOption<LogFilePath>();
                    LogManager.Configuration.Variables["logdir"] = logFilePath.Value;

                    var maximumFileAgeInDays = optionsService.GetOption<MaximumFileAge>();
                    LogManager.Configuration.Variables["maximumFileAgeInDays"] = ( (int) maximumFileAgeInDays.Value ).ToString();

                    LogManager.ReconfigExistingLoggers();
                    LogManager.KeepVariablesOnReload = true;
                }
'''

But with the given configuration (below) no logging is done.
If I replace maxArchiveFiles with a fixed integer-value, it works.
Any ideas ?

        target name="dynamicSchedulerModule"
                xsi:type="File"
                fileName="${var:logdir}/DynamicSchedulerModule-${shortdate}.log"
                archiveFileName="${var:logdir}/DynamicSchedulerModule-{#}.log"
                archiveEvery="Day"
                archiveNumbering="Date"
                archiveDateFormat="yyyy-MM-dd"
                maxArchiveFiles="${var:maximumFileAgeInDays}"
                layout="${defaultLayout}"

**Comment by snakefoot** [comment]
Think you are waiting for this:  #1234

Right now the NLog-Config-Variables are not that easy to work with when modifying them at runtime. In the perfect world, then it should be possible to do this:

```c#
LogManager.Configuration.Variables["maximumFileAgeInDays"] = ( (int) maximumFileAgeInDays.Value ).ToString();
LogManager.Configuration = LogManager.Configuration.Reload();
```

But it will NOT work. I was thinking to fix NLog-Config-Variables in NLog 5.0 as it will be breaking change.

As a work-around could be this until NLog 5.0 is ready:

```c#
NLog.LogManager.Configuration?.AllTargets.OfType<FileTarget>().ToList().ForEach(f => f.MaxArchiveFiles = (int) maximumFileAgeInDays.Value);
```

**Comment by snakefoot** [comment]
Closing as duplicate of #1234
